sirepo/job.py

```diff
@@ -159,11 +159,13 @@ def agent_cmd_stdin_env(cmd, env, uid, cwd=".", source_bashrc=""):
         cwd (str): directory for the agent to run in (will be created if it doesn't exist)
 
     Returns:
-        tuple: new cmd (tuple), stdin (file), env (PKDict)
+        tuple: new cmd (tuple), stdin (file), env (PKDict or None)
     """
     import os
     import tempfile
 
+    if sirepo.feature_config.cfg().trust_sh_env:
```

Comment:
Why also have this check if it is an arg? Can't the caller protect?

Reply:
The caller doesn't need to understand the abstraction. A lot is going on in side. It's not an arg.